x-pack/plugins/observability_solution/dataset_quality/public/components/dataset_quality_details/degraded_field_flyout/possible_mitigations/field_limit/message_callout.tsx

```diff
@@ -39,63 +38,68 @@ export function MessageCallout() {
 
 export function SuccessCallout() {
   const {
-    services: { application },
+    services: {
+      application,
+      share: {
+        url: { locators },
+      },
+    },
   } = useKibanaContextForPlugin();
   const { dataStreamSettings, datasetDetails } = useDatasetQualityDetailsState();
   const { name } = datasetDetails;
 
+  const componentTemplateUrl = locators.get('MANAGEMENT_APP_LOCATOR')?.useUrl({
+    componentTemplate: `${getComponentTemplatePrefixFromIndexTemplate(
+      dataStreamSettings?.indexTemplate ?? name
+    )}@custom`,
+  });
+
   const onClickHandler = useCallback(async () => {
-    await application.navigateToApp(MANAGEMENT_APP_ID, {
-      path: `/data/index_management/component_templates/${getComponentTemplatePrefixFromIndexTemplate(
-        dataStreamSettings?.indexTemplate ?? name
-      )}@custom`,
-      openInNewTab: true,
-    });
-  }, [application, dataStreamSettings?.indexTemplate, name]);
+    if (componentTemplateUrl) {
+      await application.navigateToUrl(componentTemplateUrl);
```

Comment:
Why don't we use directly the `.navigate` method from the locator? It is built-in and doesn't require using the `application` dependency.

Reply:
True, old habit.
But now i changed the logic and removed this